cristin-project/src/test/java/no/unit/nva/cristin/projects/common/CristinQueryTest.java

```diff
@@ -1,69 +1,92 @@
     private static final String PARENT_UNIT_ID = "185.90.0.0";
+    private static final String PER_PAGE = "10";
     private static final String QUERY_CRISTIN_PROJECTS_EXAMPLE_URI =
-        "https://api.cristin-test.uio.no/v2/projects?per_page=10&parent_unit_id=185.90.0.0&page=2&title=reindeer&lang=nb";
-    private static final String GET_ONE_CRISTIN_PROJECT_EXAMPLE_URI =
-        "https://api.cristin-test.uio.no/v2/projects/1234?lang=nb";
-
-    private static final String SAMPLE_SORT = "start_date";
+        "https://api.cristin-test.uio.no/v2/projects?lang=nb&page=2&parent_unit_id=185.90.0.0"
+        + "&per_page=10&title=reindeer";
+    private static final String QUERY_SAMPLE_WITH_MULTIPLE_PARAMETERS =
+        "https://api.cristin-test.uio.no/v2/projects?approval_reference_id=2017/1593&approved_by=REK&biobank=533895"
+        + "&biobank=533895&funding_source=NFR&institution=uib&keyword=nature&keyword=nature&lang=nb&levels=7&page=1"
+        + "&participant=St&participant=St&per_page=5&project_manager=st&sort=start_date&unit=184.12.60.0";
+    private static final String RANDOM_TITLE = "reindeer";
+    private static final String SAMPLE_FUNDING_SOURCE = "NFR";
     private static final String SAMPLE_INSTITUTION_ID = "uib";
-    private static final String SAMPLE_PROJECT_MANAGER = "st";
+    private static final String SAMPLE_LEVELS = "7";
     private static final String SAMPLE_PARTICIPANT = "St";
-    private static final String SAMPLE_FUNDING_SOURCE = "NFR";
+    private static final String SAMPLE_PROJECT_MANAGER = "st";
+    private static final String SAMPLE_SORT = "start_date";
     private static final String SAMPLE_UNIT = "184.12.60.0";
-    private static final String SAMPLE_LEVELS = "7";
-    private static final String QUERY_SAMPLE_WITH_MULTIPLE_PARAMETERS =
-            "https://api.cristin-test.uio.no/v2/projects?per_page=5&sort=start_date&approval_reference_id=2017/1593"
-                    + "&participant=St&institution=uib&unit=184.12.60.0&approved_by=REK&funding_source=NFR&page=1"
-                    + "&project_manager=st&keyword=nature&biobank=533895&levels=7";
-
-    private static final String KEYWORD = "nature";
-    private static final String BIOBANK = "533895";
-
-    private static final String APPROVAL_REFERENCE_ID = "2017/1593";
-    private static final String APPROVED_BY = "REK";
-
-
+    public static final QueryParameterKey[] QUERY_PARAMETER_KEYS = {PAGE_CURRENT, PAGE_ITEMS_PER_PAGE, LANGUAGE};
 
     @Test
-    void buildReturnsUriWithCustomParameterValuesWhenCustomParameterValuesAreSupplied() {
-        URI uri = new CristinQuery().withTitle(RANDOM_TITLE).withLanguage(LANGUAGE_NB).withItemsPerPage(PER_PAGE)
-            .withFromPage(FROM_PAGE).withParentUnitId(PARENT_UNIT_ID).toURI();
-        assertEquals(QUERY_CRISTIN_PROJECTS_EXAMPLE_URI, uri.toString());
+    void buildReturnsUriWithCustomParameterValuesWhenCustomParameterValuesAreSupplied() throws BadRequestException {
```

Comment:
consider in later PR to use shouldDoWhenSomething like the rest of the test names. I was given that suggestion myself some time ago

Reply:
should-use-propper-function-names-when-writing-test-procedures, got it.